backend/src/test/java/codezap/template/repository/SourceCodeRepositoryTest.java

```diff
@@ -0,0 +1,102 @@
+package codezap.template.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import codezap.category.domain.Category;
+import codezap.category.repository.CategoryRepository;
+import codezap.global.repository.JpaRepositoryTest;
+import codezap.member.domain.Member;
+import codezap.member.repository.MemberRepository;
+import codezap.template.domain.SourceCode;
+import codezap.template.domain.Template;
+
+@JpaRepositoryTest
+public class SourceCodeRepositoryTest {
+
+    @Autowired
+    private SourceCodeRepository sourceCodeRepository;
+    @Autowired
+    private MemberRepository memberRepository;
+    @Autowired
+    private CategoryRepository categoryRepository;
+    @Autowired
+    private TemplateRepository templateRepository;
+
+    private Member member1;
+    private Template template1;
+    private Category category1;
+    private SourceCode sourceCode1, sourceCode2, sourceCode3;
+
+    @BeforeEach
+    void setUp() {
+        member1 = memberRepository.save(new Member("user1@test.com", "pp", "salt1"));
+
+        category1 = categoryRepository.save(new Category("Category 1", member1));
+
+        template1 = templateRepository.save(new Template(member1, "Template 1", "Description 1", category1));
+
+        sourceCode1 = new SourceCode(template1, "SourceCode 1", "Content 1", 1);
+        sourceCode2 = new SourceCode(template1, "SourceCode 2", "Content 2", 2);
+        sourceCodeRepository.save(sourceCode1);
+        sourceCodeRepository.save(sourceCode2);
+    }
+
+    @Test
+    @DisplayName("소스코드 ID로 소스코드 조회")
+    void testFindBySourceCodeId() {
+        var result = sourceCodeRepository.fetchById(1L);
+
+        assertThat(result.getFilename()).isEqualTo("SourceCode 1");
+    }
+
+    @Test
+    @DisplayName("템플릿으로 전체 소스코드 조회")
+    void testFindAllByTemplate() {
+        var result = sourceCodeRepository.findAllByTemplate(template1);
```

Comment:
```suggestion
        var sourceCodes = sourceCodeRepository.findAllByTemplate(template1);
```
[제안]
`var` 와는 별개로 지역변수 이름이 더 직관적이었으면 좋겠네요

Reply:
`actual`이 좋을 것 같아요~

Reply:
보통 `result` 많이 쓰지 않나요?

Reply:
result도 나쁘진 않다고 생각해요. 

다만 저는 다음과 같은 이유로 actual을 써왔어요.

1. JUnit과 AssertJ의 assert() 메서드에서 파라미터 이름으로 actual을 사용함
2. 어떤 테스트에서는 여러 메서드의 결과를 사용하고, result에 해당하는 값이나 객체가 여러 개일 수도 있음. 이럴 때에도 일관성을 유지하기 위함

### JUnit
![Screenshot 2024-09-11 at 2 16 11 PM](https://github.com/user-attachments/assets/57dde172-40de-49b4-b987-09a3dbde50c7)

### AssertJ
![Screenshot 2024-09-11 at 2 15 21 PM](https://github.com/user-attachments/assets/c17a7ed8-2aa9-4f9e-80a4-935348bbe04c)

Reply:
2번이 잘 이해가 되지 않아서 그런데 조금만 더 자세히 설명해주실 수 있나요??
actual을 사용하면 객체가 여러 개일 때도 일관성을 유지할 수 있다는 게 잘 그려지지 않아서요😅

Reply:
제가 `actual` 이 아닌 `sourceCodes` 를 제안드린 이유는 변수 자체로 `Collection`, 또는 일급 컬렉션 정도의 타입임을 쉽게 추론할 수 있기 때문입니다.
`assertThat(result).hasSize(2);` 보단 `assertThat(sourceCodes).hasSize(2);` 가 더 파악하기 편하다고 느껴졌어요.
하지만 리뷰 본문에서도 말씀드린 것 처럼 가벼운 변수 명 제안이니 초롱이 편한 대로 사용하셔도 전혀 무관합니다!

Reply:
> 2번이 잘 이해가 되지 않아서 그런데 조금만 더 자세히 설명해주실 수 있나요?? actual을 사용하면 객체가 여러 개일 때도 일관성을 유지할 수 있다는 게 잘 그려지지 않아서요😅

이전에 미션 하면서 `assertAll()` 썼던 걸 혹시 기억할까요??

테스트 로직에 따라 검증해야할 결과값이 여러 개일 수도 있다는 의미였습니다!